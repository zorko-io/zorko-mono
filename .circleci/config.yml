# Javascript Node CircleCI 2.0 configuration file
#
# Check https://circleci.com/docs/2.0/language-javascript/ for more details
#
version: 2
jobs:
  do-everything:
    docker:
      - image: circleci/node:10.13.0
      - image: cypress/base:10
      - image: circleci/mongo:3.6
    parallelism: 1
#    working_directory: ~/repo
    steps:
      - checkout
      - restore_cache:
          keys:
            - cache-v3-{{ .Branch }}-{{ checksum "package.json" }}
      - run:
          name: Install Top Deps
          command: npm install
      - run:
          name: Install Sub Packages Deps
          command: npm run bootstrap
      - run:
          name: Verify Cypress
          command: npm run cy:verify
      - save_cache:
          key: cache-v3-{{ .Branch }}-{{ checksum "package.json" }}
          paths:
            - ~/.npm
            - ~/.cache
            - node_modules
#      - persist_to_workspace:
#          root: ~/
#          paths:
#            - .cache/Cypress
#      - attach_workspace:
#          at: ~/
      - run:
          name: Run API Integration
          command: npm run ci:api
      - run:
          name: Run UI Integration
          command: npm run ci:ui
workflows:
  version: 2
  build_and_test:
    jobs:
      - do-everything
#orbs:
# https://github.com/cypress-io/circleci-orb
# import Cypress orb by specifying an exact version x.y.z
# or the latest version 1.x.x using "@1" syntax
#  cypress: cypress-io/cypress@1
#  build:
#    docker:
#      - image: cypress/base:10
#      # Specify service dependencies here if necessary
#      # CircleCI maintains a library of pre-built images
#      # documented at https://circleci.com/docs/2.0/circleci-images/
#      - image: circleci/mongo:3.6
#    working_directory: ~/repo
#    steps:
#      - checkout
#
#      # Download and cache dependencies
#      - restore_cache:
#          keys:
#            - cache-{{ .Branch }}-{{ checksum "package.json" }}
#            - v1-dependencies-{{ checksum "package.json" }}
#            # fallback to using the latest cache if no exact match is found
#            - v1-dependencies-
#
#      - run: npm install
##      - run: npm run build
#
#      - save_cache:
#          paths:
#            - ~/.npm
#            - ~/.cache
#          key: cache-{{ .Branch }}-{{ checksum "package.json" }}
#
#      # run tests!
##      - run: npm run ci:api
#      - run: npm run cy
#workflows:
#  build:
#    jobs:
#      - cypress/run:
#          working_directory: 'packages/e2e-test-suite'
#          start: npm run ci:ui
#          wait-on: 'http://localhost:8080'
#          spec: cypress/integration/basic/*

