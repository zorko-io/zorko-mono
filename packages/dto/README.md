# `@zorko/dto`

Date Transfer Object to work with Zorko API

## Usage

```
const { UserDto, TokenDto } = require('@zorko-io/dto');

```
