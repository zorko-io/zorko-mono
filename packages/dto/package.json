{
  "name": "@zorko/dto",
  "version": "1.0.5-alpha.0",
  "description": "Data Transfer Objects for Zorko API",
  "author": "nesterone <igornester@gmail.com>",
  "homepage": "",
  "private": false,
  "license": "ISC",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib"
  ],
  "scripts": {
    "build": "tsc",
    "watch": "tsc --watch",
    "test:watch": "jest --config jest.config.json --watch",
    "test": "jest --config jest.config.json",
    "clean": "rimraf lib && rimraf node_modules"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@types/node": "^12.6.8",
    "faker": "^4.1.0",
    "livr": "^2.3.0",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^6.5.2",
    "yup": "^0.27.0"
  },
  "devDependencies": {
    "@types/yup": "^0.26.24",
    "jest": "^24.9.0",
    "rimraf": "^2.6.3"
  },
  "gitHead": "6d67667468be8bc801fa02b0a515463a315bf9b5"
}
