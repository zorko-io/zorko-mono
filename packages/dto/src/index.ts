export * from './auth';
export * from './roles';
export * from './users';
export * from './user-profile';
export * from './repository';
export * from './repository-preview';
