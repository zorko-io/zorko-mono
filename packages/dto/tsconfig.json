{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "declaration": true
  },
  "allowJs": true,
  "include": [
    "./src"
  ]
}
