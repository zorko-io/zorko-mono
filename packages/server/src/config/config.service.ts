import * as dotenv from 'dotenv';
import * as fs from 'fs';

enum ConfigKeys {
  API_AUTH_ENABLED = 'API_AUTH_ENABLED'
}

export class ConfigService {
  private readonly envConfig: { [key: string]: string };

  constructor(filePath: string = `.env.${process.env.NODE_ENV}`) {
    if (fs.existsSync(filePath)) {
      this.envConfig = dotenv.parse(fs.readFileSync(filePath));
    } else {
      this.envConfig = {};
    }

    // TODO: validate config
  }

  get(key: string): string {
    return this.envConfig[key];
  }

  get isAuthEnabled(): boolean {
    let apiAuthEnableFlag = process.env[ConfigKeys.API_AUTH_ENABLED];
    if (!apiAuthEnableFlag) {
      apiAuthEnableFlag = this.envConfig[ConfigKeys.API_AUTH_ENABLED];
    }

    if (apiAuthEnableFlag === 'true') {
      return true;
    } else if (apiAuthEnableFlag === 'false') {
      return false;
    } else {
      return true;
    }
  }
}
