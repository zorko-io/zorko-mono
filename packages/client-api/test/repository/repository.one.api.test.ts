import 'reflect-metadata';
import { ApiTestHelper } from '../helper/api.test.helper';
import { Users } from '../config';
import { repositoryFakeGenerator } from '@zorko/dto';

describe('RepositoryOneApi', () => {
   let Api;
   let actual;

  beforeEach(async () => {
    Api = await ApiTestHelper.loginAs(Users.JoeUser);
  });

  it('fails on read with wrong parameters', async () => {
      expect(() => {
        // TODO: why ts doesnt show error here ?
          Api.Repository.findOne({ zzz: 'fddfdf'})
      }).toThrowErrorMatchingSnapshot()
  });

  it('CRUD - repository (skeleton)', async () => {
    expect.assertions(3);

    let initResource = repositoryFakeGenerator.getOneRandomValidRepository({
      skipId: true
    });
    let resourceId = await Api.Repository.createOne(initResource);

    expect(resourceId).toBeTruthy();

    actual = await Api.Repository.findOne({
      id: resourceId
    });

    expect(actual).toEqual({
      ...initResource,
      id: resourceId
    });

    await Api.Repository.removeOne({ id: resourceId });

    try {
      await Api.Repository.findOne({
        id: resourceId
      });
    } catch (e) {
      expect(e.response.status).toEqual(404);
    }
  });

});
