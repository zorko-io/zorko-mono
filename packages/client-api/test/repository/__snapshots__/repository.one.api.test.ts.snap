// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryOneApi fails on read with wrong parameters 1`] = `"id is a required field"`;
